### PR TITLE
Add migration note for remote cluster settings

### PR DESCRIPTION
**Pull request by jasontedor**
The remote cluster settings `search.remote.*` have been renamed to `cluster.remote.*` and are automatically upgraded in the cluster state on gateway recovery, and on put. This commit adds a note to the migration docs for these changes.

### CODE REVIEW

File: docs/reference/migration/migrate_7_0/settings.asciidoc [hunk 1]
```diff
@@ -40,3 +40,16 @@ will be removed in the future, thus requiring HTTP to always be enabled.
 This setting has been removed, as disabling http pipelining support on the server
 provided little value. The setting `http.pipelining.max_events` can still be used to
 limit the number of pipelined requests in-flight.
+
+==== Cross-cluster search settings renamed
+
+The cross-cluster search remote cluster connection infrastructure is also used
+in cross-cluster replication. This means that the setting names
+`search.remote.*` used for configuring cross-cluster search belie the fact that
```
Comment (martijnvg) [inline docs/reference/migration/migrate_7_0/settings.asciidoc]: belie?

**Review by martijnvg — APPROVED** [review]
LGTM

**Review by s1monw — APPROVED** [review]

File: docs/reference/migration/migrate_7_0/settings.asciidoc (hunk 1 above)
Comment (jasontedor) [reply docs/reference/migration/migrate_7_0/settings.asciidoc]: It's one of my favorite words, my "lodestar" or "malign" if you will. 😇 

I use it to mean here that the names `search.remote.*` do not reveal the fact that these settings also apply to cross-cluster replication, or connections to remote clusters in general.

Comment (martijnvg) [reply docs/reference/migration/migrate_7_0/settings.asciidoc]: ok :) I did not know that

**Pull request merged**